src/client/js/RunningPluginDrawer/RunningPluginDrawer.js

```diff
@@ -0,0 +1,252 @@
+
+        this._closeBtn.on('click', function (event) {
+            event.stopPropagation();
+            event.preventDefault();
+            self.close();
+        });
+
+
+        setInterval(function () {
+            self._refreshTimes();
+        }, 1000);
+
+        client.addEventListener(CONSTANTS.CLIENT.PLUGIN_INITIATED, self.onPluginInitiated);
+        client.addEventListener(CONSTANTS.CLIENT.PLUGIN_FINISHED, self.onPluginFinished);
+    };
+
+    RunningPluginDrawer.prototype.getComponentId = function () {
+        return _getComponentId();
+    };
+
+    RunningPluginDrawer.prototype.getDefaultConfig = function () {
+        return _getDefaultConfig();
+    };
+
+    RunningPluginDrawer.prototype.getConfig = function () {
+        return _getConfig();
+    };
+
+    RunningPluginDrawer.prototype.onPluginInitiated = function (sender, event) {
+        drawer._logger.debug('plugin initiated:', {metadata: event});
+        drawer._addPluginEntry(event);
+    };
+
+    RunningPluginDrawer.prototype.onPluginFinished = function (sender, event) {
+        drawer._logger.debug('plugin finished:', {metadata: event});
+        drawer._addResultButton(event);
+        drawer._removePluginEntry(event.executionId);
+    };
+
+    RunningPluginDrawer.prototype._clear = function () {
+        this._table.empty();
+    };
+
+    RunningPluginDrawer.prototype.open = function () {
+        this._el.css('height', '250px');
+        this._isOpened = true;
+    };
+
+    RunningPluginDrawer.prototype.close = function () {
+        this._el.css('height', '0px');
+        this._isOpened = false;
+    };
+
+    RunningPluginDrawer.prototype.toggle = function () {
+        if (this._isOpened) {
+            this.close();
+        } else {
+            this.open();
+        }
+    };
+
+    RunningPluginDrawer.prototype._removePluginEntry = function (executionId) {
+        var self = this,
+            sizeWithoutMe,
+            i;
+
+        for (i = this._plugins.length - 1; i >= 0; i -= 1) {
+            if (this._plugins[i].executionId === executionId) {
+                this._plugins.splice(i, 1);
+            }
+        }
+        sizeWithoutMe = this._plugins.length;
+
+        setTimeout(function () {
+            $('#' + executionId).remove();
+            self._button.setElementNumber(sizeWithoutMe);
+            if (sizeWithoutMe === 0) {
+                self.close();
+            }
+        }, 5000);
+    };
+
+    RunningPluginDrawer.prototype._viewConfig = function (pluginEntry) {
+        var dialog = new PluginConfigDialog({client: this._client});
+
+        dialog.displayConfig(pluginEntry.metadata, pluginEntry.context.pluginConfig, function () {
+            //TODO should we even handle it? - or should we just remove from the function
+        });
+    };
+
+    RunningPluginDrawer.prototype._addPluginEntry = function (pluginEntry) {
+        var self = this,
+            entry = '<tr id="' + pluginEntry.executionId + '"><td>',
```

Comment:
The string concatenation is awkward to read. Make a separate template and/or use `$()`.